File: php-sdk/src/NewsAPIException.php
<?php

namespace AllNewsAPI;

class NewsAPIException extends \Exception
{
    private int $statusCode;

    public function __construct(string $message, int $statusCode = 0)
    {
        parent::__construct($message, $statusCode);
        $this->statusCode = $statusCode;
    }

    public function getStatusCode(): int
    {
        return $this->statusCode;
    }
}

File: java-sdk/src/main/java/com/allnewsapi/NewsAPIException.java
package com.allnewsapi;

/**
 * Custom exception for NewsAPI errors.
 * Extends RuntimeException so callers are not forced to catch it.
 */
public class NewsAPIException extends RuntimeException {
    private final int statusCode;

    /**
     * Create a new NewsAPIException.
     *
     * @param statusCode the HTTP status code
     * @param message    the error message
     */
    public NewsAPIException(int statusCode, String message) {
        super(message);
        this.statusCode = statusCode;
    }

    /**
     * Get the HTTP status code associated with this error.
     *
     * @return the HTTP status code
     */
    public int getStatusCode() {
        return statusCode;
    }
}
